app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,96 @@
+require 'date'
+
+class TasksController < ApplicationController
+
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    # Show details for one book
+    task_id = params[:id].to_i
+    @task = Task.find_by(id: task_id)
+    if @task.nil?
+      render :not_found, status: :not_found
+    end
+  end
+
+  def new
+    # Creates a new task available for new view to use
+      @task = Task.new
+  end
+
+  def create
+    # Uses post to send form data to the server/save a new book
+    @task = Task.new(task_params)
+    if @task.save
+      redirect_to root_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    # Form to edit details for an existing book.
+    # When you submit the form, it sends a patch request to update the databse
+    # Is giong to redirect before rendering new :(.
+    # Probably something to do with the if statement
+    @task = Task.find(params[:id].to_i)
```

Comment:
What happens if the task isn't found? You should include error-handling code in all of these actions that look for a single task.